drivers/disk/flashdisk.c

```diff
@@ -96,33 +135,37 @@ static int disk_flash_access_read(struct disk_info *disk, uint8_t *buff,
 }
 
 /* This performs read-copy into an output buffer */
-static int read_copy_flash_block(off_t start_addr, uint32_t size,
+static int read_copy_flash_block(struct disk_info *disk,
+				 off_t start_addr,
+				 uint32_t size,
 				 const void *src_buff,
 				 uint8_t *dest_buff)
 {
+	struct flashdisk_data *ctx;
 	off_t fl_addr;
 	uint32_t num_read;
 	uint32_t offset = 0U;
 
+	ctx = CONTAINER_OF(disk, struct flashdisk_data, info);
+
 	/* adjust offset if starting address is not erase-aligned address */
-	if (start_addr & (CONFIG_DISK_FLASH_ERASE_ALIGNMENT - 1)) {
-		offset = start_addr & (CONFIG_DISK_FLASH_ERASE_ALIGNMENT - 1);
+	if (start_addr & (ctx->page_size - 1)) {
```

Comment:
I know you didn't add the `if` here, just replaced there value, but is the `if` really required?

I think just having the assignment `offset = start_addr & (ctx->page_size - 1);` gives the same end result.